whois-api/src/main/java/net/ripe/db/whois/api/fulltextsearch/ElasticFulltextSearch.java

```diff
@@ -230,33 +233,35 @@ private String getHighlightTag(final String format, final String highlightPost)
     }
 
     private QueryStringQueryBuilder getQueryBuilder(final String query) {
-        return QueryBuilders.queryStringQuery(escape(query)).type(MultiMatchQueryBuilder.Type.PHRASE_PREFIX);
+        return QueryBuilders.queryStringQuery(escape(query)).type(MultiMatchQueryBuilder.Type.PHRASE);
     }
 
     private SearchResponse.Lst createHighlights(final SearchHit hit) {
         final SearchResponse.Lst documentLst = new SearchResponse.Lst(hit.getId());
         final List<SearchResponse.Arr> documentArrs = Lists.newArrayList();
 
-        hit.getHighlightFields().forEach((attribute, highlightField) -> {
-            if("lookup-key".equals(attribute) || "lookup-key.custom".equals(attribute)){
+        hit.getHighlightFields().values().stream().collect(getHighlightsCollector()).forEach((attribute, highlightField) -> {
+            if("lookup-key".equals(attribute)){
                 return;
             }
-            if(attribute.contains(".custom")) {
-                final SearchResponse.Arr arr = new SearchResponse.Arr(StringUtils.substringBefore(highlightField.name(), ".custom"));
-                arr.setStr(new SearchResponse.Str(null, StringUtils.join(highlightField.getFragments(), ",")));
-                documentArrs.add(arr);
-
-                //Somehow if searched term contains "." highlight field custom has no vlue for it.
-            } else if(!hit.getHighlightFields().containsKey(attribute + ".custom"))  {
-                final SearchResponse.Arr arr = new SearchResponse.Arr(highlightField.name());
-                arr.setStr(new SearchResponse.Str(null, StringUtils.join(highlightField.getFragments(), ",")));
-                documentArrs.add(arr);
-            }
+
+            LOGGER.info("final highlights are : {}:  {} - {}", attribute, highlightField.name(),   StringUtils.join(highlightField.getFragments(), ","));
```

Comment:
Should we turn this log into debug?
We will have a lot of logs if not.

Reply:
I will remove the logging, I added it for testing purpose